packages/web-app/src/controllers/appeal-details/application-decision-date.spec.js

```diff
@@ -0,0 +1,168 @@
+  const appealDecisionDate = '2021-11-19T00:00:00.000Z';
+
+  beforeEach(() => {
+    req = mockReq;
+    res = mockRes();
+    jest.clearAllMocks();
+  });
+
+  describe('getApplicationDecisionDate', () => {
+    it('should render the view with data correctly', () => {
+      req = {
+        session: {
+          appeal: {
+            appealDecisionDate,
+          },
+        },
+      };
+
+      getApplicationDecisionDate(req, res);
+
+      expect(res.render).toBeCalledTimes(1);
+      expect(res.render).toBeCalledWith(views.applicationDecisionDate, {
+        ...expectedViewData,
+        appealData: {
+          decisionDate: {
+            day: '19',
+            month: '11',
+            year: 2021,
+          },
+        },
+      });
+    });
+  });
+
+  describe('postApplicationDecisionDate', () => {
+    const newDecisionDateDay = '20';
+    const newDecisionDateMonth = '10';
+    const newDecisionDateYear = 2021;
+    const newDecisionDate = `${newDecisionDateYear}-${newDecisionDateMonth}-${newDecisionDateDay}`;
+
+    it('should call save methods with the correct data', async () => {
+      req = {
+        body: {
+          'decision-date': newDecisionDate,
+          'decision-date-day': newDecisionDateDay,
+          'decision-date-month': newDecisionDateMonth,
+          'decision-date-year': newDecisionDateYear,
+        },
+        session: {
+          appeal: {
+            appealId,
+          },
+        },
+      };
+
+      await postApplicationDecisionDate(req, res);
+
+      expect(saveAndContinue).toBeCalledTimes(1);
+      expect(saveAndContinue).toBeCalledWith({
+        req,
+        res,
+        currentPage: views.applicationDecisionDate,
+        nextPage: `${appealDetails}/${appealId}`,
+        viewData: {
+          [hasAppealSubmission.decisionDate]: new Date(`${newDecisionDate}T12:00:00.000Z`),
+        },
+        saveData: saveAppealSubmissionData,
+      });
+    });
+
+    it('should handle errors correctly', async () => {
+      req = {
+        body: {
+          'decision-date': newDecisionDate,
+          'decision-date-day': newDecisionDateDay,
+          'decision-date-month': newDecisionDateMonth,
+          'decision-date-year': newDecisionDateYear,
+          errors: {
+            key: 'value',
+          },
+        },
+        session: {
+          appeal: {
+            appealId,
+          },
+        },
+      };
+
+      await postApplicationDecisionDate(req, res);
+
+      expect(res.render).toBeCalledTimes(1);
+      expect(res.render).toBeCalledWith(views.applicationDecisionDate, {
+        ...expectedViewData,
+        appealData: {
+          decisionDate: {
+            day: newDecisionDateDay,
+            month: newDecisionDateMonth,
+            year: newDecisionDateYear,
+          },
+        },
+        errorSummary: [],
+        errors: {
+          key: 'value',
+        },
+      });
+    });
+
+    it('should handle exceptions correctly', async () => {
```

Comment:
We shouldn't need a test to cover this scenario as `saveAndContinue()` handles this scenario and already has tests to cover it.